Added modal bottom sheet that shows list of users and their reaction

E-m-i-n-e-n-c-e opened the pull request:
# See who left an emoji reaction #740

This PR adds a new feature that shows detailed information about message reactions when long-pressing a reaction button. Users can see who reacted with which emoji in a  bottom sheet.

Closes #740 


## Feature Description

When a user long-presses on a reaction:
1. A bottom sheet appears showing:
   - A row of filterable reaction buttons at the top
   - A list of users who reacted below

2. Each user row shows:
   - User's profile picture
   - User's name
 
3. Tapping on a user's row navigates to their profile

## Implementation Details

The implementation consists of two main parts:
1. `ReactionUsersSheet` - The bottom sheet UI component
2. Long-press handler in `ReactionChip` - Triggers the bottom sheet

ReactionUserSheet is in test\widgets\reaction_users_sheet_test.dart

Tests are in test\widgets\reaction_users_sheet_test.dart

 

## Testing

1. `displays emoji buttons correctly`
   - Verifies emoji buttons display with correct counts


2. `displays user list correctly`
   - Verifies user names are displayed


3. `handles unknown users gracefully`
   - Tests fallback text for unknown users


4. `navigates to user profile on tap`
   - Tests navigation to profile page


5. `switches between reactions`
   - Tests reaction switching functionality
   - Verifies user list updates



6. `handles horizontal overflow with many reactions`
   - Tests horizontal scrolling behavior
   - Verifies emoji button visibility

7. `handles vertical overflow with many users`
   - Tests vertical scrolling behavior
   - Verifies user list scrolling

8. `handles long user names without overflow`
    - Tests text overflow handling


9. `displays different types of emojis correctly`
    - Tests Unicode emoji display
    - Tests custom/image emoji display
    - Tests text emoji display

## Screenshots
 
![image](https://github.com/user-attachments/assets/b6c0f7aa-76c4-4ca6-bfae-dc9e442ec967)

![image](https://github.com/user-attachments/assets/5f28091a-8584-4651-968f-6fb337b589a2)

![image](https://github.com/user-attachments/assets/cfac2c3a-7104-4a1c-94ca-f9d869e34a3a)

E-m-i-n-e-n-c-e commented:
## Do tell me if you need any changes to the design or something like that. I think i should change the flexible widget to an expanded so that it works something like this

![image](https://github.com/user-attachments/assets/b7207677-e83b-4eae-bd52-fd9caf1fa076)

PIG208 commented:
@E-m-i-n-e-n-c-e, Thanks for working on this! We do have an issue for this feature: #740. It specifies the exact design we want for the feature. However, this is a Post-Launch issue so we might not spend as much time on implementing it.

gnprice commented:
Chat thread: https://chat.zulip.org/#narrow/channel/516-mobile-dev-help/topic/view-reactions.20sheet/near/2043883

E-m-i-n-e-n-c-e commented:
> @E-m-i-n-e-n-c-e, thanks for working on this! We do have an issue for this feature: #740. It specifies the exact design we want for the feature. However, this is a Post-Launch issue so we might not spend as much time on implementing it.

Updated the design to match the figma design

![image](https://github.com/user-attachments/assets/a184d462-a7bb-42ba-a2b2-1249603ebdfb)

![image](https://github.com/user-attachments/assets/cc82dd13-b97f-4880-83ce-e4051d19a429)


> Chat thread: https://chat.zulip.org/#narrow/channel/516-mobile-dev-help/topic/view-reactions.20sheet/near/2043883

Also added 11 tests in  test\widgets\reaction_users_sheet_test.dart

1. `displays emoji buttons correctly`
   - Verifies emoji buttons display with correct counts
   - Checks selection state

2. `displays user list correctly`
   - Verifies user names are displayed
   - Checks sorting and layout

3. `handles unknown users gracefully`
   - Tests fallback text for unknown users
   - Verifies graceful error handling

4. `navigates to user profile on tap`
   - Tests navigation to profile page
   - Verifies correct user ID passing

5. `switches between reactions`
   - Tests reaction switching functionality
   - Verifies user list updates

6. `displays online status indicator correctly`
   - Verifies green dot appears for online users
   - Checks dot positioning
   - Tests absence for offline users

7. `updates online status when user status changes`
   - Tests dynamic status updates
   - Verifies UI updates when status changes

8. `handles horizontal overflow with many reactions`
   - Tests horizontal scrolling behavior
   - Verifies emoji button visibility

9. `handles vertical overflow with many users`
   - Tests vertical scrolling behavior
   - Verifies user list scrolling

10. `handles long user names without overflow`
    - Tests text overflow handling
    - Verifies layout stability

11. `displays different types of emojis correctly`
    - Tests Unicode emoji display
    - Tests custom/image emoji display
    - Tests text emoji display

gnprice commented:
It looks like there's still quite a bit of work this PR will need in order to match all the details in the design in Figma, and in order to get the code up to our standards.

There's also an existing PR #1243 for this issue, which was already open when this one was sent and also is somewhat closer to ready.

So let's leave this PR aside. @E-m-i-n-e-n-c-e I recommend looking for another issue to try, [focusing on](https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/Flutter.20beta.20timeline/near/2078556) those in the launch milestones.
